### ISSUE TITLE
Add support for downloading subtitles in other languages

### ISSUE THREAD

**Issue by divad12** [fle-internal/te-sushi-chef#3]
Currently no subtitles are downloaded -- it's not possible to know from TE's website which languages are available. Not sure if it's possible to get this info from YouTube's API. Another thing to try is just downloading a hardcoded list of languages and allowing some to fail.

**Comment by ivanistheone** [comment]
@divad12 I haven't played with `ricecooker` subs much yet, so there might be a way to "auto extract"...

If you want to do on the chef side, you can set some `params` flags and can get a list of all avail:

```
import youtube_dl
from youtube_dl.extractor import YoutubeIE

te_url = 'https://www.youtube.com/watch?v=kEopQeg9uWU'
    
def get_all_avail_substitles(video_url):
    ydl_opts = {}
    ydl = youtube_dl.YoutubeDL(ydl_opts)
    ie = YoutubeIE()
    ydl.add_info_extractor(ie)
    ydl.params['allsubtitles'] = True
    ydl.params['writesubtitles'] = True # because otherwise subs won't be gotten...
    info_dict = ydl.extract_info(video_url, download=False)
    # print(info_dict.keys())
    # print(info_dict['extractor'])
    return info_dict['subtitles']


get_all_avail_substitles(te_url)
```
produces

```
[youtube] kEopQeg9uWU: Downloading webpage
[youtube] kEopQeg9uWU: Downloading video info webpage
[youtube] kEopQeg9uWU: Extracting video information
[youtube] kEopQeg9uWU: Downloading MPD manifest

{'en': [{'ext': 'ttml',
   'url': 'https://www.youtube.com/api/timedtext?lang=en&v=kEopQeg9uWU&fmt=ttml&name=+via+Dotsub'},
  {'ext': 'vtt',
   'url': 'https://www.youtube.com/api/timedtext?lang=en&v=kEopQeg9uWU&fmt=vtt&name=+via+Dotsub'}],
 'fr': [{'ext': 'ttml',
   'url': 'https://www.youtube.com/api/timedtext?lang=fr&v=kEopQeg9uWU&fmt=ttml&name=+via+Dotsub'},
  {'ext': 'vtt',
   'url': 'https://www.youtube.com/api/timedtext?lang=fr&v=kEopQeg9uWU&fmt=vtt&name=+via+Dotsub'}]}
```

**Comment by ralphiee22** [comment]
I have played around with the youtube api to figure out if there are subtitles in a specific language for a video. 
https://developers.google.com/youtube/v3/docs/captions/list
Using their api will return all languages that there are subtitles for. The bad thing is though that they put a limit on the number of api requests per day. One has to go through an application process in order to get more requests per day. I have plans to do this application, but I was wondering if ya'll think we should stick to using `youtube_dl` for figuring out whether there are subtitles. 
How fast is `youtube_dl` for retrieving subtitles? @ivanistheone

**Comment by ivanistheone** [comment]
I only did limited tests with one or two videos and on those videos metadata retreival seemed pretty fast (same as the usual speed for scraping video formats and info).

What's not clear to me if the languages returned from `youtube_dl` are auto-generated or real human  generated. Perhaps the API will make that clearer?

**Comment by divad12** [comment]
This has now been resolved by #5 and #8
